### PR TITLE
Add user login

### PR DESCRIPTION
**Pull request by jackrootstrap**
#### Trello board reference:

* [Trello Card #14 User login](https://trello.com/c/5EaTmjIA/32-apiary-documentation)
* [Trello Card #15 User logout](https://trello.com/c/9HwZ2TUP/15-2-as-a-user-i-should-be-able-to-logout-so-i-can-no-longer-be-logged-in)
* [Trello Card #16 User confirmation](https://trello.com/c/VJUN2VuO/16-3-as-a-user-i-should-be-able-to-confirm-my-account-after-signing-up-in-order-to-ensure-that-all-the-users-email-are-valid)
---

#### Description:

* Add specs for user login/logout session.

#### Notes:

* Add specs for login request
* Overrided sessions controller
* Updated registration controller
* Updated locals
* Created helper methods for testing

### CODE REVIEW

File: spec/requests/sessions/sign_in_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      end
+
+      it 'returns confirmation account message' do
+        expect(errors.first).to include I18n.t('api.errors.confirmation_email')
+      end
+    end
+
+    context 'when user has not confirmed the account' do
+      before do
+        user.confirm
```
Comment (sebastiancaraballo) [inline spec/requests/sessions/sign_in_spec.rb:36]: Is the confirm feature tested? Just asking

File: spec/requests/sessions/sign_out_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Sign up', type: :request do
```
Comment (sebastiancaraballo) [inline spec/requests/sessions/sign_out_spec.rb]: 👀

**Review by sebastiancaraballo — APPROVED** [review]

File: spec/requests/sessions/sign_in_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      it 'returns unauthorized status' do
+        expect(response).to have_http_status(:unauthorized)
+      end
+
+      it 'returns confirmation account message' do
+        expect(errors.first).to include I18n.t('api.errors.confirmation_email')
+      end
+    end
+
+    context 'when user has not confirmed the account' do
```
Comment (JulianPasquale) [inline spec/requests/sessions/sign_in_spec.rb]: I think this is the correct description, right?
```suggestion
    context 'when user has confirmed the account' do
```

File: spec/requests/sessions/sign_out_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  let(:sign_out_url) { '/auth/sign_out' }
+  let(:user) { create(:user) }
+
+  subject { delete sign_out_url, headers: auth_headers }
+
+  context 'User signed out' do
+    it 'returns ok status' do
+      subject
```
Comment (JulianPasquale) [inline spec/requests/sessions/sign_out_spec.rb]: Would be nice to always follow the same pattern for calling the subject:
```ruby
it '...' do
  subject
  expect ...
end
```
or
```ruby
before { subject }
```

File: spec/requests/sessions/sign_in_spec.rb (hunk 1 above)
Comment (jackrootstrap) [reply spec/requests/sessions/sign_in_spec.rb:36]: There is another [ticket](https://trello.com/c/VJUN2VuO/16-3-as-a-user-i-should-be-able-to-confirm-my-account-after-signing-up-in-order-to-ensure-that-all-the-users-email-are-valid) for that!

Comment (sebastiancaraballo) [reply spec/requests/sessions/sign_in_spec.rb:36]: Well, then that feature shouldn't be enabled yet 🤔

**Review by JulianPasquale — APPROVED** [review]

**Pull request merged**